### PR TITLE
Remove unsued and old code

### LINKED ISSUE #40: Remove python2 compatibility code
From a brief perusal of the codebase, it looks like there's still some code designed for python2 compatibility (such as unnecessary `__future__` imports). It would be good to remove this, to avoid confusion for contributors.

### PR DESCRIPTION
**Pull request by clavedeluna**
Hi there! I've worked in the python security space before and came across this project. I wanted to start helping and figured the best way is to look around and do some low-hanging fruit cleanup of code.

I found:
- one location where a util func is unused
- code that supported old py versions. I see dlint only supports 3.6+.

should also close #40

### CODE REVIEW

File: dlint/util.py [hunk 1]
```diff
@@ -1,21 +1,11 @@
 #!/usr/bin/env python
 
 import abc
-import sys
 
-if sys.version_info >= (3, 4):
-    ABC = abc.ABC
-else:
-    ABC = abc.ABCMeta(str('ABC'), (), {})
+ABC = abc.ABC
```
Comment (mschwager) [inline dlint/util.py:5]: Good catch :+1: although let's update the code to simply use `abc.ABC` everywhere and delete `util.ABC` now that it's just an alias for that class.

File: dlint/util.py [hunk 2]
```diff
@@ -1,21 +1,11 @@
...
 
 
 def lstartswith(l1, l2):
     if len(l2) > len(l1):
         return False
     return l1[:len(l2)] == l2
-
-
-def lendswith(l1, l2):
```
Comment (mschwager) [inline dlint/util.py:18]: Let's leave this in for now. I know it's unused, but it's a potentially useful function to have alongside `lstartswith`.

**Review by mschwager — COMMENTED** [review]
Nice work, thanks for the cleanup!

**Comment by mschwager** [comment]
Here's some additional Python version checks if you're interested in looking into those as well:

```
$ rg version_info
dlint/tree.py
70:    is_python_3_5 = sys.version_info >= (3, 5)

dlint/util.py
6:if sys.version_info >= (3, 4):

dlint/linters/bad_input_use.py
25:        is_python_2 = sys.version_info < (3, 0)

dlint/linters/twisted/yield_return_statement.py
24:        is_python_3_3 = sys.version_info >= (3, 3)

tests/test_bad_input_use.py
8:IS_PYTHON_2 = sys.version_info < (3, 0)

tests/test_helpers/test_bad_name_attribute_use.py
8:IS_PYTHON_3_5 = sys.version_info >= (3, 5)

tests/test_twisted/test_returnvalue_in_inlinecallbacks.py
8:IS_PYTHON_3_8 = sys.version_info >= (3, 8)

tests/test_twisted/test_inlinecallbacks_yield_statement.py
8:IS_PYTHON_3_8 = sys.version_info >= (3, 8)

tests/test_bad_exec_use.py
8:IS_PYTHON_3 = sys.version_info >= (3, 0)

tests/test_twisted/test_yield_return_statement.py
8:IS_PYTHON_3_3 = sys.version_info >= (3, 3)

tests/test_bad_re_catastrophic_use.py
10:IS_PYTHON_3_7 = sys.version_info >= (3, 7)
```

Any checks for Python older than 3.6 (the currently supported oldest version) can probably be deleted.

**Comment by clavedeluna** [comment]
```
dlint/tree.py:    is_python_3_6 = sys.version_info >= (3, 6)
tests/test_bad_re_catastrophic_use.py:IS_PYTHON_3_7 = sys.version_info >= (3, 7)
tests/test_helpers/test_bad_name_attribute_use.py:IS_PYTHON_3_6 = sys.version_info >= (3, 6)
tests/test_twisted/test_inlinecallbacks_yield_statement.py:IS_PYTHON_3_8 = sys.version_info >= (3, 8)
tests/test_twisted/test_returnvalue_in_inlinecallbacks.py:IS_PYTHON_3_8 = sys.version_info >= (3, 8)
tests/test_twisted/test_yield_return_statement.py:IS_PYTHON_3_6 = sys.version_info >= (3, 6)
```

which is what we want since dlint supports py3.6+ until #51 is complete

File: tests/test_util.py [hunk 3]
```diff
@@ -7,8 +7,11 @@
 
 class TestUtil(unittest.TestCase):
 
-    def test_abc(self):
-        assert dlint.util.ABC
+    def test_lendswith(self):
+        assert not dlint.util.lendswith([1, 2], [1, 2, 3])
+        assert not dlint.util.lendswith([1, 2, 3], [1, 2])
+
+        assert dlint.util.lendswith([1, 2, 3], [2, 3])
```
Comment (clavedeluna) [inline tests/test_util.py:14]: I added this test because even if code isn't used in repo, it should be tested :)

**Review by mschwager — APPROVED** [review]
Great work! Thank you so much!

**Pull request merged**